notebooks/samples/Regression - Vowpal Wabbit vs. LightGBM vs. Linear Regressor.ipynb

```diff
@@ -0,0 +1,390 @@
+    "from mmlspark.train import ComputeModelStatistics\n",
+    "from mmlspark.vw import VowpalWabbitRegressor, VowpalWabbitFeaturizer\n",
+    "from mmlspark.lightgbm import LightGBMRegressor\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from pyspark.ml.feature import VectorAssembler\n",
+    "from pyspark.ml.regression import LinearRegression\n",
+    "import pyspark.sql.types as T\n",
+    "from sklearn.datasets import load_boston"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prepare Dataset\n",
+    "We use [*Boston house price* dataset](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html) because of the ease of access and simplicity which is good to start with. The data was collected in 1978 from Boston area and consists of 506 entries with 14 features including the value of homes. We use `sklearn.datasets` module to download it easily, then split the set into training and testing by 75/25."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "boston = load_boston()\n",
+    "\n",
+    "feature_cols = ['f' + str(i) for i in range(boston.data.shape[1])]\n",
+    "header = ['target'] + feature_cols\n",
+    "df = spark.createDataFrame(\n",
+    "    pd.DataFrame(data=np.column_stack((boston.target, boston.data)), columns=header)\n",
+    ")\n",
+    "display(df.limit(10).toPandas())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data, test_data = df.randomSplit([0.75, 0.25], seed=42)\n",
+    "train_data.cache()\n",
+    "test_data.cache()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Following is the summary of the training set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "display(train_data.summary().toPandas())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Plot feature distributions over different target values (house prices in our case)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "features = train_data.columns[1:]\n",
+    "ncols = 5\n",
+    "nrows = math.ceil(len(features) / ncols)\n",
+    "\n",
+    "yy = [r['target'] for r in train_data.select('target').collect()]\n",
+    "\n",
+    "f, axes = plt.subplots(nrows, ncols, sharey=True, figsize=(30,10))\n",
+    "f.tight_layout()\n",
+    "\n",
+    "for irow in range(nrows):\n",
+    "    axes[irow][0].set_ylabel('target')\n",
+    "    for icol in range(ncols):\n",
+    "        try:\n",
+    "            feat = features[irow*ncols + icol]\n",
+    "            xx = [r[feat] for r in train_data.select(feat).collect()]\n",
```

Comment:
its much more efficient to select the columns in spark, then call `.toPandas` on the resulting dataframe. This will then make it in a nice form for plotting